examples/slack-security-findings/daily_findings.py

```diff
@@ -0,0 +1,188 @@
+            return False
+
+    @staticmethod
+    def is_valid_webhook(url):
+        return url is not None and url.startswith('https://hooks.slack.com/services')
+
+
+def filter_finding(finding: Finding) -> bool:
+    return date.today() - finding['first_seen_snapshot_date'] < timedelta(days=8)
+
+
+def process_findings(all_findings: Any, include: Callable[[Finding], bool]) -> list[Finding]:
+    result = []
+    for raw_finding in all_findings:
+        finding: Finding = {
+            'href': raw_finding['href'],
+            'first_seen_snapshot_date': date.fromisoformat(raw_finding['firstSeenSnapshotDate']),
+            'file_path': raw_finding['filePath'],
+            'start_line': raw_finding['startLine'],
+            'end_line': raw_finding['endLine'],
+            'type': raw_finding['type'],
+            'severity': raw_finding['severity'],
+            'severity_score': raw_finding['severityScore'],
+            'status': raw_finding['status']
+        }
+        if include(finding):
+            result.append(finding)
+    if len(result) == 0:
+        return result
+    else:
+        return sorted(result, key=lambda x: (x['severity_score'], x['first_seen_snapshot_date']), reverse=True)
+
+
+def get_filename(file_path: str | None) -> str:
+    if not file_path:
+        return ''
+    else:
+        if file_path.endswith('/'):
+            file_path = file_path[:-1]
+        parts = file_path.split('/')
+        return parts[-1]
+
+
+def create_message(system: str, findings: list[Finding], num_findings: int) -> str:
+    if len(findings) == 0:
+        message = f'No new open findings found in {args.system} in the last week! 🎉'
+    else:
+        message = f'{len(findings)} new open findings in _{system}_ during the last week'
+    if len(findings) > 5:
+        message += '. The 5 most severe ones are:\n'
+    else:
+        message += '.\n'
+    for finding in findings[:5]:
+        message += f'• [*{finding["severity"]}*, {finding["status"]}] <{finding["href"]}|{finding["type"]}> in `{get_filename(finding["file_path"])}`, '
+        if finding["start_line"] != finding["end_line"]:
+            message += f'lines {finding["start_line"]}-{finding["end_line"]}\n'
+        else:
+            message += f'line {finding["start_line"]}\n'
+    if num_findings > len(findings):
+        message += f'There are {num_findings - len(findings)} older open findings.'
+    return message
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser(description='Gets open security/reliability findings and post them to Slack.')
+    parser.add_argument('--customer', type=str, help="Name of your organization's Sigrid account.")
+    parser.add_argument('--system', type=str, help='Name of your system in Sigrid, letters/digits/hyphens only.')
+    args = parser.parse_args()
+
+    if None in [args.customer, args.system]:
+        parser.print_help()
+        sys.exit(1)
+
+    if sys.version_info.major == 2 or sys.version_info.minor < 9:
```

Comment:
I think this stuff is going to be repeated across every example, but I fear that's actually worth it to keep everything as self-contained and portable as possible.